client/trino-jdbc/src/main/java/io/trino/jdbc/NonRegisteringTrinoDriver.java

```diff
@@ -54,12 +70,16 @@ public Connection connect(String url, Properties info)
             return null;
         }
 
-        TrinoDriverUri uri = TrinoDriverUri.create(url, info);
-
-        OkHttpClient.Builder builder = httpClient.newBuilder();
-        uri.setupClient(builder);
-
-        return new TrinoConnection(uri, instrumentClient(builder.build()));
+        try {
+            TrinoDriverUri uri = TrinoDriverUri.createDriverUri(url, info);
+            OkHttpClient.Builder httpClientBuilder = HttpClientFactory.toHttpClientBuilder(uri, USER_AGENT);
+            httpClientBuilder.connectionPool(pool);
+            httpClientBuilder.dispatcher(dispatcher);
+            return new TrinoConnection(uri, instrumentClient(httpClientBuilder.build()));
+        }
+        catch (RuntimeException e) {
+            throw new SQLException(e.getMessage(), e);
```

Comment:
Is it safe to assume all runtime exceptions are `SQLException`? Why can't the throw `SQLException` in the first place?